js/language_server/src/server.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file connects LSP.js and VSCode's LSP api together
+ *
+ * We could manage all IO related things, like reading from stdin and writing to stdout, here
+ * but since we're already in JS lang, let's just use Microsofts Official TM LSP library.
+ *
+ * this allows us to use this file through node-ipc which is the defacto way to do this stuff
+ *
+ */
+import { LSFactory, LS } from "./lsp";
+
+import { createConnection, Connection } from "vscode-languageserver/node";
+
+const connection: Connection = createConnection();
+var server: LS | undefined = undefined;
+var idCntr = 0;
+connection.onInitialize(async (params) => {
+  // this is kind of gross to do this here, but it's simple.
+  server = await LSFactory();
+  // Tell the LS to use this function to write to stdout
+  server.setWriteRef((json) => {
+    // Parse JSON from ocaml code
+    const packet = JSON.parse(json);
+    // extract method and params since that's what the LSP expects
+    const method = packet.method;
+    const params = packet.params;
```

Comment:
if you wanna be a cool javascripter you could do:
```
const { method, params } = packet;
```

Reply:
noted for next time, but too lazy to fix rn